### ISSUE TITLE
Default .eslintrc.js files to node environment

### ISSUE THREAD

**Issue by Standard8** [eslint/eslint#13008]
**The version of ESLint you are using.**

eslint 6.8.0, and eslint 7.0.0-alpha.2

**The problem you want to solve.**

With ESLint 7 now defaulting to linting .eslintrc.js files (which is a good change), our repository has all the .eslintrc.js files failing due to not being in the node environment, e.g. 

```
  3:1  error  'module' is not defined.
```
Our default environment is "browser".

A work around is to use overrides:

```
  "overrides": [{
    "files": [
      ".eslintrc.js",
    ],
    "env": {
      "node": true,
      "browser": false,
    },
  }],
```

**Your take on the correct solution to problem.**

I think that ESLint should default to the node environment for its own configuration files, since that is what it expects, and isn't likely to change. This would avoid the need for repositories with a non-node default environment to specify the override.

**Are you willing to submit a pull request to implement this change?**

Maybe, if I can get guidance on how to do it.

**Comment by kaicataldo** [comment]
We don't currently have any special cases like this. I tend to lean towards not adding any unless it's absolutely critical, because as these add up, it makes increasingly difficult to understand what is actually going on (a problem we already face in our configuration).

There is one more alternative to overrides, which is an [`env` comment](https://eslint.org/docs/user-guide/configuring#specifying-environments). This is what I've used in my work as a front-end dev for repos that only contained code running in a browser.

Example:
```js
// .eslintrc.js

/* eslint-env node */
module.exports = {};
```

**Comment by Standard8** [comment]
To expand on my reasoning, as ESLint now defaults to linting its own config files (well, at least `.eslintrc.js`), then it would seem to make sense that ESLint treats them with the expected environment. Not every project that uses ESLint is a node-based project.

Using the comment is possible, though we have probably 100-200 `.eslintrc.js` files in our repository (must work on getting that down), so doing the overrides work around is the slightly better option.

I'm quite happy for this not to be incorporated as the work around is easy, I was offering it as a suggestion as it would make things simpler for non-node projects.

**Comment by kaicataldo** [comment]
Understood, thanks. I'm personally still of the mind that we shouldn't add any magic behavior here, but let's see what the rest of the team says.
